# DP: argparse.py: Make the gettext import conditional

Index: b/Lib/argparse.py
===================================================================
--- a/Lib/argparse.py
+++ b/Lib/argparse.py
@@ -87,7 +87,16 @@ import os as _os
 import re as _re
 import sys as _sys
 
-from gettext import gettext as _, ngettext
+try:
+    from gettext import gettext as _, ngettext
+except ImportError:
+    def _(message):
+        return message
+    def ngettext(singular,plural,n):
+        if n == 1:
+            return singular
+        else:
+            return plural
 
 SUPPRESS = '==SUPPRESS=='
 
