# DP: Don't run the test suite in random order.

Index: b/Tools/scripts/run_tests.py
===================================================================
--- a/Tools/scripts/run_tests.py
+++ b/Tools/scripts/run_tests.py
@@ -34,7 +34,6 @@ def main(regrtest_args):
     args.extend(['-W', 'error::BytesWarning'])
 
     args.extend(['-m', 'test',    # Run the test suite
-                 '-r',            # Randomize test order
                  '-w',            # Re-run failed tests in verbose mode
                  ])
     if sys.platform == 'win32':
